wgsl/index.bs

```diff
@@ -1366,86 +1375,360 @@ Therefore, a data value must not be placed in the padding at the end of a struct
+        // The 'const' declaration is for type vec2<f32>, so the declaration
+        // statement requires the initializer to be of type vec2<f32>.
+        // The Load Rule applies (because no other type rule can apply), and
+        // the evaluation of the initializer yields the vec2<f32> value loaded
+        // from the memory locations referenced by m[2] at the time the declaration
+        // is executed.
+        const p_m_col2: vec2<f32> = m[2];
+
+        var A: array<i32,5>;
+        // When evaluating 'A[4]'
+        // 1. First evaluate 'A', yielding a reference to the storage for
+        //    the 'A' variable. The result has type ptr<function,array<i32,5>>.
+        // 2. Then apply the '[4]' array index phrase, yielding a reference to
+        //    the storage for the fifth element of the array referenced by
+        //    the reference value from the previous step.
+        //    The result value has type ref<function,i32>.
+        // The const declaration requires the right-hand-side to be of type i32.
+        // Apply the Load Rule, so the initializer evaluates to the i32 value loaded
+        // from the memory reference computed in step 2, at the time the the
+        // declaration is executed.
+        const p_A_4: ptr<function,i32> = A[4];
+
+        // When evaluating 'person.weight'
+        // 1. First evaluate 'person', yielding a reference to the storage for
+        //    the 'person' variable declared at module scope.
+        //    The result has type ref<private,S>.
+        // 2. Then apply the '.weight' member access phrase, yielding a reference to
+        //    the storage for the second member of the memory referenced by
+        //    the reference value from the previous step.
+        //    The result has type ref<private,f32>.
+        // The const declaration requires the right-hand-side to be of type f32.
+        // Apply the Load Rule, so the initializer evaluates to the i32 value loaded
+        // from the memory reference computed in step 2, at the time the the
+        // declaration is executed.
+        const person_weight_reference: ptr<private,f32> = person.weight;
+    }
+  </xmp>
+</div>
 
-Note: Assignment of swizzled values is not permitted (SubaccessSwizzle).<br>
-           e.g. `vec4<i32> v; v.xz = vec2<i32>(0, 1);` is not allowed.
+A pointer value is formed in one of the following ways:
 
-### Pointer Evaluation TODO ### {#pointer-evaluation}
+* Use the unary '&' operator on a reference.  See #address-of-expr.
+    * The originating variable of the result is defined as the originating variable of the reference.
+* If a function [=formal parameter=] has pointer type, then when the function is invoked
+    at runtime the uses of the formal parameter denote the pointer value
+    provided to the corresponding operand at the call site in the calling function.
+    * The originating variable of the formal parameter (at runtime) is defined as
```

Comment:
From reading this and elsewhere call site deserves a definition somewhere that calls out that the call site itself is also a dynamic context.

Reply:
I added a TODO in the "Function calls" section for this.